### PR TITLE
[Relay, Quantization, TOPI] int8 dense on CUDA & Dense op quantization 

### PR DESCRIPTION
**Pull request by vinx13**
* Quantize dense layers in quantization pass
* Add `out_dtype` to `DenseAttrs` to support mixed precision.
* Add Int8 dense schedule on CUDA

@eqy @icemelon9 @ZihengJiang

### CODE REVIEW

**Comment by tqchen** [comment]
cc @masahi @nishi-t @kazum can you help review this PR?

**Comment by vinx13** [comment]
@merrymercy @yzhliu could you also help review?

File: python/tvm/relay/quantize/quantize.py [hunk 1]
```diff
@@ -150,6 +151,9 @@ def qconfig(**kwargs):
         Whether add stop_fusion when casting to dtype_activation. stop_fusion forces lowbit
         results to be stored in memory.
 
+    quantize_dense: boolean
+        Whether to quantize dense layers.
```
Comment (yzhliu) [inline python/tvm/relay/quantize/quantize.py]: maybe a better way is like that in autotvm, i.e., `quantize_op = [relay.conv2d, relay.dense]`?

Comment (vinx13) [reply python/tvm/relay/quantize/quantize.py]: @yzhliu updated

**Comment by tqchen** [comment]
@yzhliu, please https://docs.tvm.ai/contribute/code_review.html#approve-and-request-changes-explicitly

**Comment by vinx13** [comment]
@ZihengJiang @tqchen can you also take a look?

File: python/tvm/relay/quantize/quantize.py [hunk 2]
```diff
@@ -74,7 +74,8 @@ class QConfig(NodeBase):
         "round_for_shift": True,
         "store_lowbit_output": True,
         "debug_enabled_ops": None,
-        "use_stop_fusion": True
+        "use_stop_fusion": True,
+        "quantize_op": [_op.get("nn.conv2d"), _op.get("nn.dense")]
```
Comment (ZihengJiang) [inline python/tvm/relay/quantize/quantize.py]: Remove this. If you want to partially enable some ops, it can be achieved by `debug_enabled_ops`.

Comment (vinx13) [reply python/tvm/relay/quantize/quantize.py]: @ZihengJiang Done

**Comment by ZihengJiang** [comment]
Merged, thanks for the hard-working

**Pull request merged**